compiler/test/TestCompilerOnExamples.kt

```diff
@@ -1,80 +1,127 @@
  */
+//@Disabled("to save some time")
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class TestCompilerOnExamples {
-    val workingDir = Path("").absolute()    // Note: Path(".") does NOT work..!
-    val examplesDir = workingDir.resolve("../examples")
-    val outputDir = workingDir.resolve("build/tmp/test")
+    private val examplesDir = workingDir.resolve("../examples")
 
-    @Test
-    fun testDirectoriesSanityCheck() {
-        assertEquals("compiler", workingDir.fileName.toString())
-        assertTrue(examplesDir.isDirectory(), "sanity check; should be directory: $examplesDir")
-        assertTrue(outputDir.isDirectory(), "sanity check; should be directory: $outputDir")
+    @BeforeAll
+    fun setUp() {
+        sanityCheckDirectories("compiler")
+        assumeDirectory(examplesDir)
     }
 
     // TODO: make assembly stage testable - in case of failure (eg of 64tass) it Process.exit s
 
-    fun testExample(nameWithoutExt: String, platform: ICompilationTarget, optimize: Boolean) {
-        val filepath = examplesDir.resolve("$nameWithoutExt.p8")
-        val result = compileProgram(
-            filepath,
-            optimize,
-            writeAssembly = true,
-            slowCodegenWarnings = false,
-            compilationTarget = platform.name,
-            libdirs = listOf(),
-            outputDir
-        )
-        assertTrue(result.success, "${platform.name}, optimize=$optimize: \"$filepath\"")
+    private fun makeDynamicCompilerTest(name: String, platform: ICompilationTarget, optimize: Boolean) : DynamicTest {
+        val searchIn = mutableListOf(examplesDir)
+        if (platform == Cx16Target) {
+            searchIn.add(0, examplesDir.resolve("cx16"))
+        }
+        val filepath = searchIn.map { it.resolve("$name.p8") }.first { it.exists() }
+        val displayName = "${examplesDir.relativize(filepath)}: ${platform.name}, optimize=$optimize"
+        return dynamicTest(displayName) {
+            compileProgram(
+                filepath,
+                optimize,
+                writeAssembly = true,
+                slowCodegenWarnings = false,
+                compilationTarget = platform.name,
+                libdirs = listOf(),
+                outputDir
+            ).assertSuccess("; $displayName")
+        }
     }
 
+    @TestFactory
+    @Disabled
+    fun bothCx16AndC64() = mapCombinations(
+        dim1 = listOf(
+            "animals",
+            "balls",
+            "cube3d",
+            "cube3d-float",
+            "cube3d-gfx",
+            "cxlogo",
+            "dirlist",
+            "fibonacci",
+            "line-circle-gfx",
+            "line-circle-txt",
+            "mandelbrot",
+            "mandelbrot-gfx",
+            "numbergame",
+            "primes",
+            "rasterbars",
+            "screencodes",
+            "sorting",
+            "swirl",
+            "swirl-float",
+            "tehtriz",
+            "test",
```

Comment:
examples/test.p8 should not be included -- I often put temporary code in it that might not compile at all (just to debug some things for instance).  I already fixed this in your source branch.